src/modeler/AutoRest.Swagger/Validation/Core/Rule.cs

```diff
@@ -25,14 +25,14 @@ protected Rule()
         /// <summary>
         /// What kind of open api document type this rule should be applied to
         /// </summary>
-        public virtual ServiceDefinitionDocumentType ServiceDefinitionDocumentType => ServiceDefinitionDocumentType.ARM;
+        public abstract ServiceDefinitionDocumentType ServiceDefinitionDocumentType { get; }
```

Comment:
A. Defaults make us lazy and open the door to rely on them when adding new rules.
B. When explicitly having to set these properties we will end up doing our due diligence and understand the impact and implications of the validation rule.